pkg/controller/operators/catalog/operator.go

```diff
@@ -1383,19 +1390,50 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 	// Does the namespace have an operator group that specifies a user defined
 	// service account? If so, then we should use a scoped client for plan
 	// execution.
-	kubeclient, crclient, err := o.clientAttenuator.AttenuateClientWithServiceAccount(plan.Status.AttenuatedServiceAccountRef)
+	kubeclient, crclient, dynamicClient, err := o.clientAttenuator.AttenuateClientWithServiceAccount(plan.Status.AttenuatedServiceAccountRef)
 	if err != nil {
 		o.logger.Errorf("failed to get a client for plan execution- %v", err)
 		return err
 	}
 
-	ensurer := newStepEnsurer(kubeclient, crclient)
+	ensurer := newStepEnsurer(kubeclient, crclient, dynamicClient)
 
 	for i, step := range plan.Status.Plan {
 		switch step.Status {
 		case v1alpha1.StepStatusPresent, v1alpha1.StepStatusCreated:
 			continue
+		case v1alpha1.StepStatusWaitingForAPI:
+			switch step.Resource.Kind {
+			case crdKind:
+				crd, err := o.opClient.ApiextensionsV1beta1Interface().ApiextensionsV1beta1().CustomResourceDefinitions().Get(step.Resource.Name, metav1.GetOptions{})
+				if err != nil {
+					if k8serrors.IsNotFound(err) {
+						plan.Status.Plan[i].Status = v1alpha1.StepStatusNotPresent
+					} else {
+						return errorwrap.Wrapf(err, "error finding the %s CRD", crd.Name)
+					}
+					continue
+				}
 
+				established, namesAccepted := false, false
```

Comment:
This can wait for a follow up, but I think the CRD readiness can be wrapped up into a helper function to clean this up some.